pycaret/internal/PycaretExperiment.py

```diff
@@ -1965,6 +2011,74 @@ def setup(
             display,
         )
 
+        if not self._is_unsupervised():
+            from sklearn.model_selection import (
+                StratifiedKFold,
+                KFold,
+                GroupKFold,
+                TimeSeriesSplit,
+            )
+
+            from sktime.forecasting.model_selection import (
+                ExpandingWindowSplitter,
+                SlidingWindowSplitter,
+            )
+
+            fold_random_state = self.seed if self.fold_shuffle_param else None
+
+            if fold_strategy == "kfold":
+                self.fold_generator = KFold(
+                    self.fold_param,
+                    random_state=fold_random_state,
+                    shuffle=self.fold_shuffle_param,
+                )
+            elif fold_strategy == "stratifiedkfold":
+                self.fold_generator = StratifiedKFold(
+                    self.fold_param,
+                    random_state=fold_random_state,
+                    shuffle=self.fold_shuffle_param,
+                )
+            elif fold_strategy == "groupkfold":
+                self.fold_generator = GroupKFold(self.fold_param)
+            elif fold_strategy == "timeseries":
+                self.fold_generator = TimeSeriesSplit(self.fold_param)
+            elif (fold_strategy == "expandingwindow") or (fold_strategy == "slidingwindow"):
+                if isinstance(self.data_before_preprocess, pd.DataFrame):
+                    y_size = len(self.y_train) # self.data_before_preprocess.size
+                elif isinstance(self.data_before_preprocess, pd.Series):
+                    y_size = len(self.y_train) # self.data_before_preprocess.size
+                else:
+                    raise TypeError("data parameter must be a pandas.Series or pandas.DataFrame.")
+
+                # window_length = fh * (self.fold_param - 1)
+                window_length = len(fh)
+                step_length = len(fh)
+                initial_window = y_size - (self.fold_param * window_length)
+                #window_length = y_size - window_length
+
+                if initial_window < 1:
+                    raise ValueError("Not Enough Data Points, set a lower number of folds or fh")
+
+                if fold_strategy == "expandingwindow":
+                    self.fold_generator = ExpandingWindowSplitter(
+                        initial_window=initial_window,
+                        step_length=step_length,
+                        window_length=window_length,
+                        fh=fh,
+                        start_with_window=True
+                    )
+
+                if fold_strategy == "slidingwindow":
```

Comment:
perhaps here is missing an `elif`. 
```python
# First case 
if fold_strategy == "expandingwindow":
    ...
# Second case
elif fold_strategy == "slidingwindow":
```

Reply:
Since only 1 strategy is passed, this should work.